dockerfiles/geospatial-dev-osgeo_4.2.0.Dockerfile

```diff
@@ -6,7 +6,7 @@ LABEL org.opencontainers.image.licenses="GPL-2.0-or-later" \
       org.opencontainers.image.authors="Carl Boettiger <cboettig@ropensci.org>"
 
 ENV PROJ_VERSION=9.0.0
-ENV GDAL_VERSION=3.5.0
+ENV GDAL_VERSION=3.5.0.1
```

Comment:
GDAL seems to be using 4-digit TAGs as well as this one, but they have not issued a release for 4-digit tags. (i.e. I think the build of dev-osgeo Docker image will fail trying to download a file that doesn't exist: GDAL 3.5.0.1)
The tag detection logic needs to be changed...